On my machine the fast tests fail:
```
> just fast-tests
leloup@leloup-Legion-7-16ITHg6:~/repositories/espressosys/optimism-espresso-integration$ just fast-tests > test.log
./run_fast_tests.sh
forge fmt || true
just forge-build-dev 
FOUNDRY_PROFILE=lite forge build 
just interfaces-check-no-build
go run ./scripts/checks/interfaces
go test -v   ./...
# github.com/ethereum-optimism/optimism/op-batcher/batcher
batcher/service.go:15:2: "github.com/ethereum/go-ethereum/common" imported and not used
error: Recipe `go_test` failed on line 24 with exit code 1
error: Recipe `fast-tests` failed on line 6 with exit code 1

```

Fix looks easy though.

--- a/espresso/streamer.go
+++ b/espresso/streamer.go
@@ -9,8 +9,10 @@ import (
 	"time"
 
 	espressoClient "github.com/EspressoSystems/espresso-network-go/client"
+	espressoLightClient "github.com/EspressoSystems/espresso-network-go/light-client"
 	espressoTypes "github.com/EspressoSystems/espresso-network-go/types"
 	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
@@ -45,6 +47,7 @@ type EspressoStreamer struct {
 
 	L1Client            L1Client // TODO Philippe apparently not used yet
 	EspressoClient      *espressoClient.Client
+	EspressoLightClient *espressoLightClient.LightClientReader
 	Log                 log.Logger
 
 	// Batch number we're to give out next
@@ -75,28 +78,29 @@ func (s *EspressoStreamer) Refresh(ctx context.Context, syncStatus *eth.SyncStat
 		return false, nil
 	}
 
-	hotshotLatestBlockHeight, err := s.EspressoClient.FetchLatestBlockHeight(ctx)
+	hotshotState, err := s.EspressoLightClient.LightClient.
+		FinalizedState(&bind.CallOpts{BlockNumber: new(big.Int).SetUint64(syncStatus.SafeL2.L1Origin.Number)})
 	if err != nil {
 		return false, err
 	}
 
 	s.confirmedBatchPos = syncStatus.SafeL2.Number
-	s.confirmedHotShotPos = hotshotLatestBlockHeight
+	s.confirmedHotShotPos = hotshotState.BlockHeight
 	s.Reset()
 	return true, nil
 }
 
 func (s *EspressoStreamer) Update(ctx context.Context) error {
 	// Fetch more batches from HotShot if available.
-	hotshotLatestBlockHeight, err := s.EspressoClient.FetchLatestBlockHeight(ctx)
+	hotshotState, err := s.EspressoLightClient.LightClient.FinalizedState(&bind.CallOpts{})
 
 	if err != nil {
 		return fmt.Errorf("failed to fetch HotShot block height: %w", err)
 	}
 
-	s.Log.Debug("Updated finalized hotshot state", "hotshotLatestBlockHeight", hotshotLatestBlockHeight)
+	s.Log.Debug("Updated finalized hotshot state", "hotshotState", hotshotState)
 
-	targetHeight := min(hotshotLatestBlockHeight, s.hotShotPos+100)
+	targetHeight := min(hotshotState.BlockHeight, s.hotShotPos+100)
 
 	for ; s.hotShotPos < targetHeight; s.hotShotPos += 1 {
 		s.Log.Debug("fetching HotShot block", "blockNr", s.hotShotPos)

--- a/op-batcher/batcher/service.go
+++ b/op-batcher/batcher/service.go
@@ -12,7 +12,6 @@ import (
 
 	espresso "github.com/EspressoSystems/espresso-network-go/client"
 	opcrypto "github.com/ethereum-optimism/optimism/op-service/crypto"
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"

